docs/developing-operators.md

```diff
@@ -84,6 +84,75 @@ Now your first operator is ready and you can install it to cluster. You can do i
 
 For simplicity if you want to install what we created here without actually replicating it on your filesystem your can just clone KUDO repository and then run `kubectl kudo install ./config/samples/first-operator`.
 
+## Operator.yaml file
+
+This is the main piece of every operator. It consists of two main parts. First one defines metadata about your operator.
+
+```yaml
+name: operator
+description: my first super awesome operator
+version: "5.7"
+kudoVersion: ">= 0.2.0"
+kubeVersion: ">= 1.14"
+maintainers:
+  - Bob <bob@example.com>
+  - Alice <alice@example.com>
+url: https://github.com/myoperator/myoperator
+```
+
+Most of these are provided as a form of documentation. `kudoVersion` and `kubeVersion` use semver constraints to define minimal or maximal version of kubernetes or kudo that this operator supports. Under the hood, we use [this library](https://github.com/Masterminds/semver) to evaluate the constraints.
+
+### Tasks section
+
+Another part of `operator.yaml` is tasks section. Tasks are the smallest pieces of work that get executed together. You usually group kubernetes manifests that should be applied at once into one task. An example can be a deploy task that will result in config.yaml and pod.yaml being applied to your cluster.
+
+```yaml
+tasks:
+  deploy-task:
+    resources:
+      - config.yaml
+      - pod.yaml
+```
+
+### Plans section
+
+Plans orchestrate tasks through `phases` and `steps`.
+
+Each Plan is a tree with a fixed three-level hierarchy of the plan itself, its phases, and then steps within those phases. The choice of three levels was arbitrarily chosen as “enough levels for anybody”. This three-level hierarchy can look as follows:
```

Comment:
1. I don't think the levels are arbitrary and 
2. "enough levels for anybody" seems not  supportive and doesn't seems like it needs to be in our official documentation.

The hierarch is:
1. a "plan" which is a named collection of phases and steps
2. "phases" is a collection of steps AND also provides an ability to group serial steps and concurrent steps with a point of demarcation that allows for health before moving forward.  This point of validation at the end of phase could be a human verification in an upgrade flow.
3. "Steps" is the actual action to be applied via tasks as you describe.

Reply:
I copied that sentence from SDK docs, I am fine with droping it. It does not offend me, it shows that we did some design behind the three levels and it's the case right now, we have three levels, no less no more. But I am fine droping that sentence, I don't have any personal attachment to that since it's taken from SDK docs :)

Reply:
I think we could probably drop both sentences entirely. As a reader, I think it's a brand new concept that they probably don't have an opinion on - and if they do, hopefully it's in a KEP somewhere.